.pan/drafts/pan-2230.md

```
@@ -0,0 +1,230 @@
+### WI-3 — Wire `lint:circular` into the lint chain (bead `wire-lint-circular`)
+
+In `package.json`, add the script and insert it into the chained `lint` **before** `lint:ratchet-audit` (audit runs last):
+
+```jsonc
+"lint": "eslint src/ --ext .js,.ts,.tsx --no-inline-config && npm run lint:effect && npm run lint:permissions && npm run lint:state-writes && npm run lint:overdeck-boundaries && npm run lint:skills && npm run lint:prompts && npm run lint:file-size && npm run lint:source-introspection && npm run lint:circular && npm run lint:ratchet-audit",
+"lint:circular": "bash scripts/lint-circular-deps.sh",
+```
+
+### WI-4 — Extend the ratchet audit (bead `ratchet-audit-integration`)
+
+In `scripts/lint-ratchet-audit.sh`, treat `scripts/circular-deps-baseline.txt` like the `eslint-any-allowlist.json` set-of-lines file: add a `circular_increases_for_commit()` mirroring `allowlist_increases_for_commit()` (a `comm -13` of parent vs. commit, prefix `circular cycle added: `), add a `CIRCULAR_BASELINE="scripts/circular-deps-baseline.txt"` var, include it in the final `for file in "$BASELINE" "$ALLOWLIST" "$CIRCULAR_BASELINE"` loop, and include its increases in `audit_commit`'s `increases=$(...)` block. Any added cycle line without an issue reference in the commit message fails; with one, it passes.
```

markdownlint-cli2 (0.22.1)

[warning] Spaces inside code span elements

(MD038, no-space-in-code)